### ISSUE TITLE
deprecate --extra option to trial

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#3372]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)| @exarkun reported|
|-|-|
|Trac ID|trac#3372|
|Type|task|
|Created|2008-07-29 20:23:45Z|
|Branch|https://github.com/twisted/twisted/tree/remove-trial-extra-3372|

It serves no useful purpose any longer.

<details><summary>Searchable metadata</summary>

```
trac-id__3372 3372
type__task task
reporter__exarkun exarkun
priority__low low
milestone__ 
branch__branches_remove_trial_extra_3372 branches/remove-trial-extra-3372
branch_author__cyli cyli
status__closed closed
resolution__fixed fixed
component__trial trial
keywords__easy easy
time__1217363025000000 1217363025000000
changetime__1295224164000000 1295224164000000
version__None None
owner__cyli cyli
cc__cyli cc__thijs
```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="glyph's avatar" src="https://avatars.githubusercontent.com/u/716529?s=50" width="50" height="50">](https://github.com/glyph)<a name="note_1"></a>|@glyph commented|
|-|-|

Specifically, its purpose as described in its usage documentation is now accomplished by the `--testmodule` option.

This ticket raises an interesting issue though.  I don't think there should ever be a 'deprecate X' ticket without a parallel 'remove X' ticket, at least until the removals are automated.  I filed [#3374](https://github.com/twisted/twisted/issues/3374) in an attempt to keep it on our radar.

(Also I'm lowering the priority because I really hope that we have more useful things to do than remove trivial options.  Don't interpret it as a signal that it should not be done though; the option is kind of silly, and I don't know of any other use-cases for it.)

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)|@thijstriemstra set owner to @cyli|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="cyli's avatar" src="https://avatars.githubusercontent.com/u/223200?s=50" width="50" height="50">](https://github.com/cyli)<a name="note_3"></a>|@cyli commented|
|-|-|

(In [30444]) Branching to 'remove-trial-extra-3372'

**Comment by twisted-trac** [comment]
|[<img alt="cyli's avatar" src="https://avatars.githubusercontent.com/u/223200?s=50" width="50" height="50">](https://github.com/cyli)<a name="note_4"></a>|@cyli commented|
|-|-|

(In [30446]) Twisted coding standard compliance

refs [#3372](https://github.com/twisted/twisted/issues/3372)

**Comment by twisted-trac** [comment]
|[<img alt="cyli's avatar" src="https://avatars.githubusercontent.com/u/223200?s=50" width="50" height="50">](https://github.com/cyli)<a name="note_5"></a>|@cyli removed owner|
|-|-|

So "remove-trial-extra" may be an inaccurate branch description.  Nevertheless... [build results](http://buildbot.twistedmatrix.com/boxes-supported?branch=/branches/remove-trial-extra-3372)

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_6"></a>|@thijstriemstra set owner to @cyli|
|-|-|

 - The man page needs a copyright update:
```
Copyright \(co 2003-2010 Twisted Matrix Laboratories
```
 - can you also update the copyright in `twisted.version.copyright`? There's not really a policy that says it should be updated when working on a ticket that touches that part of the code (like a release) so you might as well do it here.

Cheers

**Comment by twisted-trac** [comment]
|[<img alt="cyli's avatar" src="https://avatars.githubusercontent.com/u/223200?s=50" width="50" height="50">](https://github.com/cyli)<a name="note_7"></a>|@cyli commented|
|-|-|

(In [30453]) Updated several copyrights

refs [#3372](https://github.com/twisted/twisted/issues/3372)

**Comment by twisted-trac** [comment]
|[<img alt="cyli's avatar" src="https://avatars.githubusercontent.com/u/223200?s=50" width="50" height="50">](https://github.com/cyli)<a name="note_8"></a>|@cyli commented|
|-|-|

(In [30454]) Oops, one more copyright.

refs [#3372](https://github.com/twisted/twisted/issues/3372)

**Comment by twisted-trac** [comment]
|[<img alt="cyli's avatar" src="https://avatars.githubusercontent.com/u/223200?s=50" width="50" height="50">](https://github.com/cyli)<a name="note_9"></a>|@cyli removed owner|
|-|-|

Not sure what you meant by ```twisted.version.copyright```, so updated both ```twisted.python.versions``` 's copyright, and ```twisted.copyright```.

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_10"></a>|@thijstriemstra commented|
|-|-|

Replying to [cyli](#note_9):
> Not sure what you meant by ```twisted.version.copyright```, 

Sorry I meant `twisted.copyright.copyright`

> so updated both ```twisted.python.versions``` 's copyright, and ```twisted.copyright```.

even better, thanks.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_11"></a>|@exarkun commented|
|-|-|

Files that get changed should get their copyright date updated.  Random extra files should not (including `twisted/copyright.py` which will get updated during a release).

**Comment by twisted-trac** [comment]
|[<img alt="cyli's avatar" src="https://avatars.githubusercontent.com/u/223200?s=50" width="50" height="50">](https://github.com/cyli)<a name="note_12"></a>|@cyli commented|
|-|-|

so... should I change both back?

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_13"></a>|@exarkun commented|
|-|-|

If you want to, it would be preferable, but it teeters on the edge of not-worth-it.  Mainly I want to clear up whether something similar should ever be done for another ticket (no it should not).

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_14"></a>|@exarkun set owner to @cyli|
|-|-|

  1. in `ExtraTests`, `assertEquals` is preferred over `assertEqual`
  1. Existing deprecations use `Version("Twisted", ...)` rather than `Version("twisted", ...)`.  I even feel like this made a real difference at some point, but I can't quite remember what the case was.

Otherwise looks good.

**Comment by twisted-trac** [comment]
|[<img alt="cyli's avatar" src="https://avatars.githubusercontent.com/u/223200?s=50" width="50" height="50">](https://github.com/cyli)<a name="note_15"></a>|@cyli commented|
|-|-|

(In [30490]) Rollback extra copyright changes

refs [#3372](https://github.com/twisted/twisted/issues/3372)

**Comment by twisted-trac** [comment]
|[<img alt="cyli's avatar" src="https://avatars.githubusercontent.com/u/223200?s=50" width="50" height="50">](https://github.com/cyli)<a name="note_16"></a>|@cyli set status to `closed`|
|-|-|

(In [30494]) Merge remove-trial-extra-3372: deprecate --extra option to trial

Author: cyli
Reviewer: thijs, exarkun
Fixes: [#3372](https://github.com/twisted/twisted/issues/3372)

Deprecate --extra option to trial

**Comment by twisted-trac** [comment]
|[<img alt="cyli's avatar" src="https://avatars.githubusercontent.com/u/223200?s=50" width="50" height="50">](https://github.com/cyli)<a name="note_17"></a>|@cyli set status to `reopened`|
|-|-|

(In [30502]) Reverse the merge because changing 'twisted' to 'Twisted' in Versions causes a failure in one of the new tests

reopens [#3372](https://github.com/twisted/twisted/issues/3372)

**Comment by twisted-trac** [comment]
|[<img alt="cyli's avatar" src="https://avatars.githubusercontent.com/u/223200?s=50" width="50" height="50">](https://github.com/cyli)<a name="note_18"></a>|@cyli commented|
|-|-|

(In [30503]) Changed the 'twisted' in Version to 'Twisted'

refs [#3372](https://github.com/twisted/twisted/issues/3372)

**Comment by twisted-trac** [comment]
|[<img alt="cyli's avatar" src="https://avatars.githubusercontent.com/u/223200?s=50" width="50" height="50">](https://github.com/cyli)<a name="note_19"></a>|@cyli set status to `closed`|
|-|-|

(In [30506]) Merge remove-trial-extra-3372: deprecate --extra option to trial

Author: cyli
Reviewer: thijs, exarkun 
Fixes: [#3372](https://github.com/twisted/twisted/issues/3372)

Deprecate --extra option to trial
